src/deque.py

```diff
@@ -0,0 +1,57 @@
+"""Implementation of a Deque in Python."""
+
+from dll import DbLinkedList
+
+
+class Deque(object):
+    """Implementation of a deque data structure.
+
+    append(value) adds value to the end of the deque.
+    appendleft(value) adds value to the front of the deque.
+    pop() removes and returns a value from the end of the deque.
+    popleft() removes and returns a value from the front of the deque.
+    peek() returns the tail value without removing it.
+    size() returns the length of the deque.
+    """
+
+    def __init__(self, value=None):
+        """Init a deque."""
+        self._dblinkedlist = DbLinkedList()
+        if value:
+            self.append(value)
+
+    def append(self, value):
+        """Add value to the end of the deque."""
+        self._dblinkedlist.append(value)
+
+    def appendleft(self, value):
+        """Add value to the front of the deque."""
+        self._dblinkedlist.push(value)
+
+    def pop(self):
+        """Remove value from the end of the end of the deque."""
+        if self._dblinkedlist.head is None:
+            raise ValueError("Cannot pop from an empty Deque")
+        return self._dblinkedlist.shift()
+
+    def popleft(self):
+        """Remove value from the front of the deque."""
+        if self._dblinkedlist.head is None:
+            raise ValueError("Cannot popleft from an empty Deque")
+        return self._dblinkedlist.pop()
+
+    def peek(self):
+        """Return the value of the end of the deque without removing it."""
+        if self._dblinkedlist.head is None:
+            return self._dblinkedlist.head
+        return self._dblinkedlist.tail.value
+
+    def peekleft(self):
+        """Return the value of the front of the deque without removing it."""
+        if self._dblinkedlist.head is None:
+            return self._dblinkedlist.head
+        return self._dblinkedlist.head.value
```

Comment:
Under what condition does the second return statement get used and what value should it always return?